tests/controller_test/collect-data_test.go

```diff
@@ -0,0 +1,172 @@
+package controller_test
+
+import (
+	"context"
+	"github.com/litmuschaos/chaos-exporter/controller"
+	"github.com/litmuschaos/chaos-exporter/pkg/clients"
+	"github.com/litmuschaos/chaos-operator/api/litmuschaos/v1alpha1"
+	litmusFakeClientSet "github.com/litmuschaos/chaos-operator/pkg/client/clientset/versioned/fake"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes/fake"
+	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
+	"testing"
+)
+
+func TestGetResultList(t *testing.T) {
+	FakeChaosNameSpace := "Fake Namespace"
+	FakeEngineName := "Fake Engine"
+
+	tests := map[string]struct {
+		chaosresult *v1alpha1.ChaosResult
+		monitoring  *controller.MonitoringEnabled
+		isErr       bool
+	}{
+		"Test Positive-1": {
```

Comment:
instead of a generic name, can you provide a meaning test name (describe the testcase)